### PR TITLE
 fix(nested-interactive): add message for negative tabindex

### PR DESCRIPTION
**Pull request by dan-tripp**
First I split the "no-focusable-content" check into three checks, one for each rule that uses it.  That was my best guess on how to do it.  If not for #3163, then for #3163 and #2934 later.

Then I added a messageKey for "nested-interactive".

Closes issue: #3163

### CODE REVIEW

**Comment by dan-tripp** [comment]
Nice!  Thank you for these comments.  I'll get started on fixing these issues now.

**Comment by dan-tripp** [comment]
Thank you again.  I think you're doing the hard part here.  I hope my latest commit fits the bill.

**Review by straker — DISMISSED** [review]
Sorry this is taking so long. Getting our messages so they are understandable and give useful remediation help sometimes takes a lot of work. I've added some suggestions to finalize the messages.

Also, with this change we need to add unit tests and integration tests. For frame-focusable-content we can duplicate the current [no-focusable-content unit tests](https://github.com/dequelabs/axe-core/blob/develop/test/checks/keyboard/no-focusable-content.js). We'll then want to update the no-focusable-content unit tests to add a test to capture negative tabindex. The test should also check that the `data` property has the `messageKey` set to `notHidden`. (you can look at https://github.com/dequelabs/axe-core/blob/develop/test/checks/keyboard/landmark-is-top-level.js#L4-L21 for an example of how to setup the test to use `checkContext` to get the data).

Lastly, we'll want to update the [nested-interactive integration test](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/nested-interactive/nested-interactive.html) to add a failure for an element with negative tabindex, and probably another one that also adds `aria-hidden=true` just for completeness.

File: lib/checks/keyboard/no-focusable-content.json [hunk 1]
```diff
@@ -5,7 +5,10 @@
     "impact": "serious",
     "messages": {
       "pass": "Element does not have focusable descendants",
-      "fail": "Element has focusable descendants",
+      "fail": {
+        "default": "Element has focusable descendants",
+        "notHidden": "Using negative tabindex is not a reliable way of hiding interactive elements"
```
[the remark this answers is not in the document]
Comment (dan-tripp) [reply lib/checks/keyboard/no-focusable-content.json]: If I'm reading [straker's comment here](https://github.com/dequelabs/axe-core/pull/3194#discussion_r735965627) correctly, he's satisfied with [the description and other messages as of this PR in its current state](https://github.com/dan-tripp/axe-core/blob/ba1e4e65fb5be3263aec635408603b5a550ce3a4/lib/rules/nested-interactive.json).  I am under the impression that he's the pertinent code owner here, so I'm going to mark this conversation as resolved.

**Comment by dan-tripp** [comment]
@straker   Thank you for your patience.  I'm new to git, and rebase.  I think I fixed the merge problems, and that this PR is ready for your consideration again.

**Comment by straker** [comment]
We can worry about the presentation / none message in a different pr since this one's already large.

**Review by straker — APPROVED** [review]

**Comment by straker** [comment]
Reviewed for security

**Comment by dan-tripp** [comment]
Great!  Thank you again for all the help.
